types/stylelint/index.d.ts

```diff
@@ -246,7 +246,7 @@ declare namespace stylelint {
 	/** @internal */
 	export type RuleBase<P = any, S = any> = (
 		primaryOption: P,
-		secondaryOptions: Record<string, S>,
+		secondaryOptions: S,
```

Comment:
```suggestion
		secondaryOptions: S extends Record<string, any> ? S : never,
```

see https://github.com/stylelint/stylelint/pull/7950/files#r1730844881

Reply:
Sorry for my late comment. Keeping it simple (i.e., `secondaryOptions: S`) seems enough for now.